src/storage/lock_manager/mod.rs

```diff
@@ -67,9 +91,14 @@ impl From<u64> for WaitTimeout {
     }
 }
 
+#[derive(Debug, Clone)]
+pub struct KeyLockWaitInfo {
+    pub key: Key,
+    pub lock_digest: LockDigest,
+    pub lock_info: LockInfo,
+}
+
 /// Uniquely identifies a lock-waiting request in a `LockManager`.
-///
-/// Not used yet, but necessary for implementing `LockWaitQueues`.
 #[derive(Clone, Copy, Hash, PartialEq, Eq, Debug)]
 pub struct LockWaitToken(pub Option<u64>);
```

Comment:
When there is not a valid token allocated. None represents invalid.
Actually it can be a simple u64 and use `Option<LockWaitToken>` when it's possibly invalid. I'll consider update this in further PRs

Reply:
Is it possible that a token fails to be allocated?

Reply:
No, but it's possible to be not-allocated-yet.